backend/src/controllers/coachController.ts

```diff
@@ -57,3 +57,35 @@ export const getCoaches = async (req: any, res: Response) => {
     res.status(400).send(error);
   }
 };
+
+export const updateCoachOrAdmin = async (req: any, res: Response) => {
+  try {
+    const userId = req.params.id;
+
+    const { name, email, role } = req.body;
+
+    const validationResult = editUserSchema.validate({ name, email, role });
+    if (validationResult.error) {
+      console.error(validationResult);
```

Comment:
Could you return the error because this can make user register like email which is not valid or role which is not valid  so it is better if you send the error response like return res.status(400).send(validationResult.error)